packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
It can't be an iso^ return type, because we can't be sure that `A` doesn't violate the isolation boundary. For example, adding to Array[String ref] ref results in an Array[String ref] ref. The result can't be iso because it points to the same mutable strings.